tiny_dnn/core/framework/tensor_storage.h

```diff
@@ -1,152 +0,0 @@
-/*
-    Copyright (c) 2013, Taiga Nomi and the respective contributors
-    All rights reserved.
-
-    Use of this source code is governed by a BSD-style license that can be found
-    in the LICENSE file.
-*/
-#pragma once
-
-#include <memory>
-#include <type_traits>
-#include <vector>
-
-#include "tiny_dnn/core/framework/device.fwd.h"
-
-#if defined(USE_OPENCL) || defined(USE_CUDA)
-#ifdef USE_OPENCL
-#include "third_party/CLCudaAPI/clpp11.h"
-#else
-#include "third_party/CLCudaAPI/cupp11.h"
-#endif
-#endif
-
-namespace tiny_dnn {
-
-template <typename Container>
-static inline size_t product(Container &c) {
-  return std::accumulate(std::begin(c), std::end(c), size_t(1),
-                         std::multiplies<size_t>());
-}
-
-template <typename C1, typename C2>
-static inline size_t compute_offset(const C1 &start, const C2 &shape) {
-  size_t res = 0;
-  for (size_t i = 0; i < shape.size(); ++i) {
-    res *= shape[i];
-    res += (i < start.size()) ? *(start.begin() + i) : 0;
-  }
-  return res;
-}
-
-template <typename U = float_t, typename Allocator = aligned_allocator<U, 64>>
-class TensorStorage {
-  typedef typename std::vector<U, Allocator>::iterator DataIter;
-  typedef typename std::vector<U, Allocator>::const_iterator ConstDataIter;
-
- public:
-  /**
-   * Initializes an empty tensor storage.
-   * @return
-   */
-  TensorStorage() {}
-
-  /**
-   * Constructor that assepts a vector of shape and create a TensorStorage
-   * with a size equivalent to that shape.
-   * @param shape array containing N integers, sizes of dimensions
-   * @return
-   */
-  explicit TensorStorage(const std::vector<size_t> &shape) { resize(shape); }
-
-  /**
-   * Constructor that assepts an initializer list  of shape and create a
-   * TensorStorage with a size equivalent to that shape.
-   * @param shape array containing N integers, sizes of dimensions
-   * @return
-   */
-  explicit TensorStorage(std::initializer_list<size_t> const &shape) {
-    resize(shape);
-  }
-
-  /**
-   * Sychronizes data on host and device
-   */
-  void sync() {
-    if (data_dirty_ && data_is_on_host_) {
-      toDevice();
-    } else {
-      fromDevice();
-    }
-  }
-
-  /**
-   *
-   * @param offset
-   * @return iterator to an element at offset position
-   */
-  DataIter host_data(size_t offset) { return host_data_.begin() + offset; }
-
-  /**
-   *
-   * @param offset
-   * @return  constant iterator to an element at offset position
-   */
-  ConstDataIter host_data(size_t offset) const {
-    return host_data_.begin() + offset;
-  }
-
-  void resize(const std::vector<size_t> &sz) { host_data_.resize(product(sz)); }
-
-  void resize(std::initializer_list<size_t> const &sz) {
-    host_data_.resize(product(sz), U(0));
-  }
-
-  size_t size() const { return host_data_.size(); }
-
- private:
-  void toDevice() const {
```

Comment:
If it will be reused, than probably inside `Tensor`, not different class. Luckily, git stores the history, so there is no point to keep dead (as for now) code